File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/TimedAlarmReceiver.java
package com.nikhilparanjape.radiocontrol.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;

import com.nikhilparanjape.radiocontrol.services.BackgroundAirplaneService;

public class TimedAlarmReceiver extends WakefulBroadcastReceiver {
    public static final int REQUEST_CODE = 12345;
    public static final String ACTION = "com.nikhilparanjape.radiocontrol.services.BackgroundAirplaneService";

    public TimedAlarmReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        Intent i = new Intent(context, BackgroundAirplaneService.class);
        context.startService(i);
    }
}

File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/TimedAlarmReceiver.kt
@file:Suppress("DEPRECATION")

package com.nikhilparanjape.radiocontrol.receivers

import android.content.Context
import android.content.Intent
import androidx.legacy.content.WakefulBroadcastReceiver

import com.nikhilparanjape.radiocontrol.services.BackgroundJobService

/**
 * Created by Nikhil Paranjape on 08/26/2016.
 *
 * @author Nikhil Paranjape
 *
 *
 */

class TimedAlarmReceiver : WakefulBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val i = Intent(context, BackgroundJobService::class.java)
        context.startService(i)
    }

    companion object {
        const val REQUEST_CODE = 142456
    }
}
